daemon-scheduler/pkg/deployment/deployment_worker.go

```diff
@@ -120,6 +110,41 @@ func (d deploymentWorker) updateDeployment(environment *types.Environment,
 		return nil, err
 	}
 
+	return resp, nil
+}
+
+func (d deploymentWorker) updateDeployment(ctx context.Context,
+	environment *types.Environment, deployment *types.Deployment,
+	resp *ecs.DescribeTasksOutput) (*types.Deployment, error) {
+
+	updatedDeployment, err := d.updateDeploymentObject(deployment, resp)
+	if err != nil {
+		return nil, err
+	}
+
+	// retrieve in-progress again to make sure it has not been updated by another process
+	// TODO: wrap the in-progress check and updateDeployment in a transaction
+	deployment, err = d.deployment.GetInProgressDeployment(ctx, environment.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	if deployment == nil || deployment.ID != updatedDeployment.ID {
```

Comment:
This is in the case where the in-progress deployment has already been modified. I can add a log statement here.